ledgerwallet/manifest_json.py

```diff
@@ -0,0 +1,60 @@
+import json
+import os
+from typing import Optional
+
+from ledgerwallet import params
+from ledgerwallet.manifest import AppManifest, icon_from_file
+from ledgerwallet.utils import get_device_name
+
+
+class AppManifestJson(AppManifest):
+    def __init__(self, filename):
+        with open(filename) as f:
+            self.path = os.path.dirname(filename)
+            self.dic = json.load(f)
+            assert "targetId" in self.dic and "binary" in self.dic
+
+    @property
+    def app_name(self) -> str:
```

Comment:
This method can be removed if it just forward to the mother class samely-named method. Inheritance will deal with it for free.